client/src/component/sos/audit/RiskLimitForm.tsx

```diff
@@ -8,63 +10,117 @@ interface FormProps {
 }
 
 interface FormState {
+    ballotPollingField: string;
     ballotPollingLimit: number;
+    comparisonField: string;
     comparisonLimit: number;
 }
 
 
 const DEFAULT_RISK_LIMIT = 0.05;
+const MIN_RISK_LIMIT = 0.0001;
+const MAX_RISK_LIMIT = 1 - MIN_RISK_LIMIT;
+
+
+function isValidRiskLimit(limit: number) {
+    return isFinite(limit)
+        && MIN_RISK_LIMIT <= fromPercent(limit)
+        && fromPercent(limit) <= MAX_RISK_LIMIT;
+}
+
+function fromPercent(val: number) {
+    return val / 100;
+}
+
+function toPercent(val: number) {
+    return val * 100;
+}
 
 
 class RiskLimitForm extends React.Component<FormProps & any, FormState> {
     public state: FormState = {
+        ballotPollingField: `${toPercent(DEFAULT_RISK_LIMIT)}`,
         ballotPollingLimit: DEFAULT_RISK_LIMIT,
+        comparisonField: `${toPercent(DEFAULT_RISK_LIMIT)}`,
         comparisonLimit: DEFAULT_RISK_LIMIT,
     };
 
     public render() {
-        const { ballotPollingLimit, comparisonLimit } = this.state;
+        const {
+            ballotPollingField,
+            ballotPollingLimit,
+            comparisonField,
+            comparisonLimit,
+        } = this.state;
 
         this.props.forms.riskLimit = this.state;
 
         return (
             <div>
                 <div>
                     <label>
-                        Ballot Polling Audits
+                        Ballot Polling Audits (%)
                         <NumericInput
-                            minorStepSize={ 0.001 }
-                            stepSize={ 0.01 }
-                            value={ ballotPollingLimit }
+                            allowNumericCharactersOnly={ true }
+                            min={ toPercent(MIN_RISK_LIMIT) }
+                            max={ toPercent(MAX_RISK_LIMIT) }
+                            minorStepSize={ toPercent(0.001) }
+                            onBlur={ this.onBlur }
+                            stepSize={ toPercent(0.01) }
+                            value={ ballotPollingField }
                             onValueChange={ this.onBallotPollingValueChange } />
                     </label>
                 </div>
                 <div>
                     <label>
-                        Comparison Audits
+                        Comparison Audits (%)
                         <NumericInput
-                            minorStepSize={ 0.001 }
-                            stepSize={ 0.01 }
-                            value={ comparisonLimit }
+                            allowNumericCharactersOnly={ true }
```

Comment:
Does this constraint permit one to type a decimal point?

Reply:
@kiniry yes it does.

Reply:
👍